processing/src/main/java/org/apache/druid/query/groupby/GroupByStatsProvider.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.groupby;
+
+import org.apache.druid.guice.LazySingleton;
+import org.apache.druid.query.groupby.epinephelinae.LimitedTemporaryStorage;
+
+import javax.inject.Inject;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Collects stats for group by queries like used merged buffer count, spilled bytes and group by resource acquisition time.
+ */
+@LazySingleton
+public class GroupByStatsProvider
+{
+  private final AtomicLong resourceAcquisitionTimeNs = new AtomicLong(0);
+  private final AtomicLong resourceAcquisitionCount = new AtomicLong(0);
```

Comment:
```suggestion
  private final AtomicLong mergeBufferAcquisitionTimeNs = new AtomicLong(0);
  private final AtomicLong mergeBufferAcquisitionCount = new AtomicLong(0);
```

Reply:
Making them atomic seems redundant given that both need to be updated together, and guaranteeing atomicity over a single variable is not useful. The current code does not rely on the `AtomicLong` part of the variables since they are already guarded by the provider's lock.